core/discovery/reducer/all.go

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package reducer
```

Comment:
According to functional programming terms, reducer is an operator which takes two or more values and returns a single one. Sum or multiplication are reducers. I.e. list (1 ,2 ,3 ,4).Reduce(Sum) = 10. What your code does here is simply filtering collection values according to some predicates. Maybe it's worth to call that package - predicate/filter?